lib/request.js

```diff
@@ -9,6 +10,12 @@ module.exports = (options) => {
     return new Promise((resolve, reject) => {
         const chunks = [];
         const request = httpModule.request(options, (response) => {
+            response.on('error', reject);
+            const encoding = response.headers && response.headers['content-encoding'];
+            if (encoding === 'gzip' || encoding === 'deflate') {
+                response = response.pipe(zlib.createUnzip());
```

Comment:
probably it's better to create new variable instead of reassignment parameter. I know that works now, so it's just about code style. what do you think about it, @shuhei ?